fcrepo-indexing-solr/src/main/java/org/fcrepo/camel/indexing/solr/SolrRouter.java

```diff
@@ -136,48 +136,41 @@ public void configure() throws Exception {
                 .setHeader(HTTP_QUERY).simple("commitWithin=" + config.getCommitWithin())
                 .to(config.getSolrBaseUrl() + "/update?useSystemProperties=true");
 
-        from("direct:external.ldpath").routeId("FcrepoSolrLdpathFetch")
-                .removeHeaders("CamelHttp*")
-                .setHeader(HTTP_URI).header(INDEXING_TRANSFORMATION)
-                .setHeader(HTTP_METHOD).constant("GET")
-                .to("http://localhost/ldpath");
-
-        from("direct:transform.ldpath").routeId("FcrepoSolrTransform")
-                .removeHeaders("CamelHttp*")
-                .setHeader(HTTP_URI).simple(config.getLdpathServiceBaseUrl())
-                .setHeader(HTTP_QUERY).simple("context=${headers.CamelFcrepoUri}")
-                .to("http://localhost/ldpath");
 
         /*
          * Handle update operations
          */
         from("direct:update.solr").routeId("FcrepoSolrUpdater")
                 .log(LoggingLevel.INFO, logger, "Indexing Solr Object ${header.CamelFcrepoUri}")
-                .setBody(constant(null))
                 .setHeader(INDEXING_URI).simple("${header.CamelFcrepoUri}")
                 // Don't index the transformation itself
                 .filter().simple("${header.CamelIndexingTransformation} != ${header.CamelIndexingUri}")
                 .choice()
-                .when(header(INDEXING_TRANSFORMATION).startsWith("http"))
-                .log(LoggingLevel.INFO, logger,
-                        "Fetching external LDPath program from ${header.CamelIndexingTransformation}")
-                .to("direct:external.ldpath")
-                .setHeader(HTTP_METHOD).constant("POST")
-                .to("direct:transform.ldpath")
-                .to("direct:send.to.solr")
-                .when(or(header(INDEXING_TRANSFORMATION).isNull(), header(INDEXING_TRANSFORMATION).isEqualTo("")))
-                .setHeader(HTTP_METHOD).constant("GET")
-                .to("direct:transform.ldpath")
-                .to("direct:send.to.solr")
-                .otherwise()
-                .log(LoggingLevel.INFO, logger, "Skipping ${header.CamelFcrepoUri}");
+                    .when(header(INDEXING_TRANSFORMATION).startsWith("http"))
+                        .log(LoggingLevel.INFO, logger,
+                            "Sending RDF for Transform with with XSLT from ${header.CamelIndexingTransformation}")
+                        .toD("xslt:${header.CamelIndexingTransformation}")
+                        .to("direct:send.to.solr")
+                    .when(header(INDEXING_TRANSFORMATION).isNotNull())
+                        .log(LoggingLevel.INFO, logger,
+                            "Sending RDF for Transform with with XSLT from ${header.CamelIndexingTransformation}")
+                        .toD("xslt:${header.CamelIndexingTransformation}")
+                        .log(LoggingLevel.INFO, logger,"${body}")
+                        .to("direct:send.to.solr")
```

Comment:
This "when" statement and the one above it seem to perform the exact same actions except for an additional log. Could one be removed?

Reply:
I can't be sure what my thinking was on this but it looks like an extra debug I accidentally left in place. I'm merging these